src/pyosrd/schedules/schedules.py

```diff
@@ -370,7 +370,7 @@ def train_delay(self, train, ref_schedule: 'Schedule') -> pd.DataFrame:
 
         return self.delays(ref_schedule).max().loc[train]
 
-    def compute_weighted_delays(
+    def compute_weighted_delay(
```

Comment:
However this is not a propery and most of others non property function are using verb/actions.

I would suggest compute_total_weighted_delay

Reply:
After discussion the rule of thumb would be more : 
- No verb/action when the schedule in itself is not modified by the method
- verb/action can be used otherwise